### PR TITLE
fix: uniV3 pool balance calculation

### PR DESCRIPTION
**Pull request by philbow61**
### Description

This PR fixes the logic around calculating UnIV3 Pool balances.
The math comes from https://atiselsts.github.io/pdfs/uniswap-v3-liquidity-math.pdf

### Other changes

ran yarn pretty that's where the changes to other non Uniswap related files come from 

### Tested

Updated the tests.
Additionally, I checked that the numbers match the actual values by comparing them with our existing pools:
luckily these covered all three cases:
1. price below our range https://app.uniswap.org/pool/3113?chain=celo
2. price in range https://app.uniswap.org/pool/55551?chain=celo
3. price above range https://app.uniswap.org/pool/3115?chain=celo

### Related issues

https://github.com/mento-protocol/mento-core/issues/484

### CODE REVIEW

File: src/helpers/UniV3PoolBalanceCalculator.test.ts [hunk 1]
```diff
@@ -3,76 +3,87 @@ import { BigNumber } from "ethers";
...
-    if (poolAddress === this.pool1Address) return this.pool1TotalLiqudity;
-    if (poolAddress === this.pool2Address) return this.pool2TotalLiqudity;
-  }
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   public async getERC20Decimals(tokenAddress: string): Promise<number> {
+    if (tokenAddress === "0x765DE816845861e75A25fCA122bb6898B8B1282a")
+      return 18;
+    if (tokenAddress === "0x1a35EE4640b0A3B87705B0A4B45D227Ba60Ca2ad") return 8;
+    if (tokenAddress === "0xd71Ffd0940c920786eC4DbB5A12306669b5b81EF")
+      return 18;
```
Comment (nvtaveras) [inline src/helpers/UniV3PoolBalanceCalculator.test.ts:86]: maybe assign the addresses to variables for better readability? i.e. `cUSD = 0x..`

File: src/helpers/UniV3PoolBalanceCalculator.ts [hunk 2]
```diff
@@ -41,46 +41,83 @@ export class UniV3PoolBalanceCalculator {
...
-          `Pool liquidity is 0 for pool ${poolAddress} with position ${positions[i]}`,
+      const sqrtRatioLower = Math.sqrt(1.0001 ** tickLower);
+      const sqrtRatioUpper = Math.sqrt(1.0001 ** tickUpper);
+      const sqrtPrice = sqrtPriceX96.dividedBy(Q96);
+
+      let amount0 = new BigNumber(0);
+      let amount1 = new BigNumber(0);
+
+      // Formula for all cases comes from
+      // https://atiselsts.github.io/pdfs/uniswap-v3-liquidity-math.pdf
```
Comment (nvtaveras) [inline src/helpers/UniV3PoolBalanceCalculator.ts:77]: As discussed, would be great to add some comments here about the way the calculation works for the different scenarios

**Review by nvtaveras — APPROVED** [review]
Great work, verified that the returned amounts match the positions in the pools listed in the desc

File: src/helpers/UniV3PoolBalanceCalculator.test.ts (hunk 1 above)
Comment (philbow61) [reply src/helpers/UniV3PoolBalanceCalculator.test.ts]: sounds good 👍

File: src/helpers/UniV3PoolBalanceCalculator.ts (hunk 2 above)
Comment (philbow61) [reply src/helpers/UniV3PoolBalanceCalculator.ts]: Left a comment above the function explaining all the steps and formulas.
Do you think they are sufficient?

**Review by bayological — APPROVED** [review]

**Review by nvtaveras — APPROVED** [review]

**Pull request merged**